pliers/filters/text.py

```diff
@@ -49,8 +55,80 @@ def __init__(self, stemmer='porter', *args, **kwargs):
             raise ValueError("stemmer must be either a valid string, or an "
                              "instance of class StemmerI.")
         self.stemmer = stemmer
+        self.tokenize = tokenize
         super(WordStemmingFilter, self).__init__()
 
     def _filter(self, stim):
-        stemmed = self.stemmer.stem(stim.text)
+        if self.tokenize:
+            tokens = stim.text.split()
+            stemmed = ' '.join([self.stemmer.stem(tok) for tok in tokens])
+        else:
+            stemmed = self.stemmer.stem(stim.text)
         return TextStim(stim.filename, stemmed, stim.onset, stim.duration)
+
+
+class TokenizingFilter(TextFilter):
+
+    ''' Tokenizes a TextStim into several word TextStims.
+    Args:
+        tokenizer (nltk Tokenizer): a nltk Tokenizer to tokenize
+            with. Will use the word_tokenize method if none is specified.
+    '''
+
+    _log_attributes = ('tokenizer',)
+
+    def __init__(self, tokenizer=None):
+        self.tokenizer = tokenizer
+        super(TokenizingFilter, self).__init__()
+
+    def _filter(self, stim):
+        if self.tokenizer:
```

Comment:
Same suggestion as above: we could validate the type of `tokenizer`, and/or allow a string naming a valid tokenizer in the module.